src/schema/opportunity.ts

```diff
@@ -464,6 +477,47 @@ export const resolvers: IResolvers<unknown, BaseContext> = traceResolvers<
         keywords: [],
       };
     },
+    getOpportunities: async (
+      _,
+      { state }: { state?: number },
+      ctx: Context,
+      info,
+    ): Promise<GQLOpportunity[]> => {
+      // Default to LIVE opportunities if no state is provided
+      const opportunityState = state ?? OpportunityState.LIVE;
+
+      if (!ctx.userId) {
+        throw new NotFoundError('Not found!');
+      }
+
+      return await graphorm.query<GQLOpportunity>(
+        ctx,
+        info,
+        (builder) => {
+          builder.queryBuilder.where({ state: opportunityState });
+
+          // If fetching non-LIVE opportunities and user is not a team member,
+          // filter at database level to only show opportunities where user is a recruiter
+          if (opportunityState !== OpportunityState.LIVE && !ctx.isTeamMember) {
```

Comment:
Why do we apply it only if the opp isn't live? It should apply on all opps regardless i think